doc/design/object-bucket-lib.md

```diff
@@ -92,31 +92,41 @@ Another issue was that a decentralized design didn't support a reasonable separa
-1. the creation of the physical bucket with owner credentials (performed by provisioner).
-1. the creation of a Secret, based on the provisioner's returned credentials, residing in the OBC's namespace (performed by bucket lib).
+1. (greenfield) generation of a random bucket name when requested (performed by bucket lib).
+1. (greenfield) the creation of the physical bucket with owner credentials (performed by provisioner).
 1. the creation of a ConfigMap, based on the provisioner's returned OB, residing in the OBC's namespace (performed by bucket lib).
+1. the creation of a Secret, based on the provisioner's returned credentials, residing in the OBC's namespace (performed by bucket lib).
+This step usually requires the provisioner to create an object store user with the appropriate bucket access credentials.
 
 `Bound` is one of the supported phases of an OB and an OBC.
 `Bound` indicates that a bucket and all related artifacts have been created on behalf of the OBC. Once a bucket claim is bound the app pod can run, meaning the Secret (containing access credentials) and the ConfigMap (containing the bucket endpoint) are mounted and consumable by the pod.
 
 ### Bucket Deletion
-When an OBC is deleted the provisioner's `Delete()` or `Suspend()` methods will be called depending on the OB's _reclaimPolicy_.
-If the reclaim policy is "delete" then the provisioner's `Delete()` method is called and the bucket is expected to be physically removed.
-If the reclaim policy is "retain" then the provisioner's `Suspend()` method is called and the bucket is expected to not be physically removed.
-However, object store specific clean up such as deleting bucket credentials, detach, archive, etc. can be performed depending on the needs of the object store.
+For greenfield buckets, when an OBC is deleted, the provisioner's `Delete`, `Suspend`, or `Erase` methods will be called depending on the OB's _reclaimPolicy_.
+If the storage class's reclaim policy is "Delete" then the provisioner's `Delete` method is called and the bucket is expected to be physically removed.
+If the reclaim policy is "Retain" then the provisioner's `Suspend` method is called and the bucket is expected to remain with all its data (objects) intact.
+If the reclaim policy is "Erase" then the provisioner's `Erase` method is called and the bucket is expected to remain with all its data (objects) removed. This policy allows the bucket name to be preserved while deleting its contents.
+
+For brownfield buckets, when an OBC is deleted, the provisioner's `Revoke` method is called.
+The provisioner decides whether or not to recognize the reclaimPolicy.
+It is anticipated that most provisioners will chose to ignore the reclaimPolicy and simply cleanup up credentials, users, etc.
+
+In all delete cases store-specific clean up, such as deleting credentials, detach, archive, etc. can be performed by any of the delete related methods.
```

Comment:
missing comma `In all delete cases, store-specific`.  comma following `clean up` can be dropped, but isn't required